src/main/java/org/broadinstitute/hellbender/utils/LocatableComparator.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.hellbender.utils;
+
+import htsjdk.samtools.util.Locatable;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Objects;
+
+/**
+ * Parent class for {@link Locatable locatable} {@code Comparator comparators} that adheres to the standard position order
+ * within a contig.
+ *
+ * <p>
+ *     All descendant classes must respect the following order of {@link Locatable}s within a contig:
+ *     <ul>
+ *         <li>instances with a lower {@code start} position come first</li>
+ *         <li>then if two instances have the same {@code start} location the ones with a smaller {@code end} location come first.</li>
+ *     </ul>
+ * </p>
+ * <p>
+ *     Also, the {@code null} contig is considered the last contig and so all instances with a non-null contig would come before
+ *     any instances with a {@code null} contig.
+ * </p>
+ * <p>
+ *     The relative order between contig is what is left to be determined by the implementing class by defining {@link #compareContigs(String,String)}.
+ * </p>
+ */
+public abstract class LocatableComparator implements Comparator<Locatable>, Serializable {
```

Comment:
Can you explain why you need this new abstract comparator class? The previous lexicographic comparator was 3 lines of code and did the same thing:

```
    public static final Comparator<Locatable> LEXICOGRAPHICAL_ORDER_COMPARATOR =
            Comparator.comparing(Locatable::getContig,Comparator.nullsLast(String::compareTo))
                    .thenComparingInt(Locatable::getStart)
                    .thenComparingInt(Locatable::getEnd);
```